src/utils/vm_mount.cpp

```diff
@@ -58,6 +58,22 @@ mp::VMMount::VMMount(const std::string& sourcePath,
       uid_mappings(std::move(uidMappings)),
       mount_type(mountType)
 {
+    auto print_mappings = [](auto& iter_begin, const auto& iter_end) {
+        std::string retval;
+        while (iter_begin != iter_end)
+        {
+            retval += fmt::format("{}:{}; ", (*iter_begin).first, (*iter_begin).second);
+            ++iter_begin;
+        }
+        return retval.substr(0, retval.size() - 2);
+    };
+
+    if (auto unique_iter_end = mp::unique_id_mappings(gid_mappings); unique_iter_end != gid_mappings.end())
+        throw std::runtime_error(fmt::format("Mount cannot apply mapping with duplicate gids: {}",
```

Comment:
I don't think the message is very clear here.  It doesn't reference the first mapping found with the duplicate.  For example, if I passed in `-g 1000:1000 -g 1000:1001 -g 1000:10002`, it will only print out the last two.  I think it should print *all* of the duplicate mappings and let the user see what they are and decide how to fix it.

Also, I'm kind of on the fence on this, but I wonder if we should be a little more specific *which* gid is the duplicate meaning either the host gid or the instance gid that is the offender.  Any opinions on this?

Reply:
I agree that the error message could be more descriptive, but it require the filtering algorithm to change quite a bit so that the offending id can be identified as well as it's associated duplicate id mappings. I haven't thought of a good way to do this yet and seeing as how other things are higher priority I haven't done anything yet.

Reply:
Ok, makes sense.  We can put this on the back burner for now until the other items you are working on are more complete.